frontend/src/components/templates/ILab/index.jsx

```diff
@@ -0,0 +1,214 @@
+                      setExpanded(item, !isResultExpanded(item.id)),
+                    expandId: `expandId-${uid()}`,
+                  }}
+                />
+
+                <Td>{item.primary_metrics[0]}</Td>
+                <Th>{formatDateTime(item.begin_date)}</Th>
+                <Th>{formatDateTime(item.end_date)}</Th>
+                <Td>
+                  <StatusCell value={item.status} />
+                </Td>
+              </Tr>
+              <Tr isExpanded={isResultExpanded(item.id)}>
+                <Td colSpan={8}>
+                  <ExpandableRowContent>
+                    <Accordion asDefinitionList={false} togglePosition="start">
+                      <AccordionItem>
+                        <AccordionToggle
+                          onClick={() => {
+                            onToggle("bordered-toggle1");
+                          }}
+                          isExpanded={expanded.includes("bordered-toggle1")}
+                          id="bordered-toggle1"
+                        >
+                          Metadata
+                        </AccordionToggle>
+
+                        <AccordionContent
+                          id="bordered-expand1"
+                          isHidden={!expanded.includes("bordered-toggle1")}
+                        >
+                          <div className="metadata-wrapper">
+                            <Card className="metadata-card" isCompact>
+                              <CardBody>
+                                <MetaRow
+                                  key={uid()}
+                                  heading={"Fields"}
+                                  metadata={[
+                                    ["benchmark", item.benchmark],
+                                    ["name", item.name],
+                                    ["email", item.email],
+                                    ["source", item.source],
+                                  ]}
+                                />
+                              </CardBody>
+                            </Card>
+                            <Card className="metadata-card" isCompact>
+                              <CardBody>
+                                <MetaRow
+                                  key={uid()}
+                                  heading={"Tags"}
+                                  metadata={Object.entries(item.tags)}
+                                />
+                              </CardBody>
+                            </Card>
+                            <Card className="metadata-card" isCompact>
+                              <CardBody>
+                                <MetaRow
+                                  key={uid()}
+                                  heading={"Common Parameters"}
+                                  metadata={Object.entries(item.params)}
+                                />
+                              </CardBody>
+                            </Card>
```

Comment:
This has been bothering me for a while, but I'm not quite sure what I want to do about it. "Parameters" are really attached to an iteration, not a run. The "run parameters" are the subset that's identical for all iterations. In the ilab training runs, there's a single iteration. In the ilab sdg runs, there are 5 iterations which vary only in the "batch-size" parameter.

So this "shortcut" code is fine for training (when we got started), but starts looking a bit odd for sdg runs since it'll omit the batch size entirely, and there's no way in the UI to see the various iteration values.

One possibility is a set of iteration accordions that can be opened, but that seems awkward to use and read. On the other hand, I fear it'll be a mess if we just show them all.

Then again, we'll be adding a lot of tags as well to capture all sorts of metadata, and the simple table of tag values may become equally awkward. We might need to think about a better presentation model for all of these...

![image](https://github.com/user-attachments/assets/9735a0e7-f58b-4410-8f62-7377f7533057)

Reply:
Each run is a separate row in the top-level view -- I'm talking about the expansion view of an individual run. Each run is either `train` or `sdg` or whatever; but while the `train` runs currently have a single iteration (and therefore just one set of parameter values), `sdg` runs currently have 5 iterations with different `block-size` parameter values (all the others are the same ... currently). So the question is how do we display the various iterations of one run, without making the expanded view too busy.

I don't like having to scroll down to see the graph, but one possibility would be just to rearrange the cards and put the graph first (if there's any agreement on *that* point), and let the iteration data "go on forever" at the bottom.

The tag values (which are associated with a run rather than individual iterations) are going to be fairly critical as that's how Crucible will represent GPU hardware, RHEL AI and InstructLab versions, etc.; I think we want *that* at the top, although I'd like to minimize the vertical size somehow, and it probably doesn't need the full width.

Reply:
I think this discussion would benefit from a call.

Reply:
Yes; let's think about "next steps" once we've got the comparison view up and we can all sit and stare at what we've got. 😀